### ISSUE TITLE
Implement LLM explainations

### ISSUE THREAD

**Issue by Benoss** [hsnborn22/LinGo#13]
LLM are now really good at language, being able to have a hotkey to send a prompt with the word (or the sentence) would be nice.

I can imagine this working with a local text file representing the prompt and using the go template language have access to the source language, the destination language, the highlighted word, the current sentence as part of the context. Render text then send the prompt to the configured AI model. 

This would obviously be bring your own key, having support for openAI, anthropic, google, mixtral and ollama models would be cool

**Comment by hsnborn22** [comment]
Yes it's a really cool idea; in these next weeks I'll be very busy but as soon as I become more free I can start working on it; if you want to contribute and need any help with code explanation you can ask here.
Regarding the feature, there are a few implementation problems, but they're not that big of a deal:

- You would have to launch the python server for every language you learn if you wish to use this feature, and it would probably be a little uncomfortable.
- LLMs are still very bad at some languages (e.g. arabic or amharic). In the first case this is because while fusha arabic is rather standard, no one really uses it irl and tons of non standardized dialects (which are often non mutually intelligible) exist. So if someone used this app to learn arabic for example, this feature would not turn out to be so useful.
